source/Halibut.Tests/Support/SerilogLoggerBuilder.cs

```diff
@@ -105,5 +115,35 @@ public void Emit(LogEvent logEvent)
                 }
             }
         }
+
+        public class TraceLogsForFailedTestsSink : ILogEventSink
+        {
+            private readonly MessageTemplateTextFormatter _formatter;
+
+            public TraceLogsForFailedTestsSink(MessageTemplateTextFormatter formatter) => _formatter = formatter != null ? formatter : throw new ArgumentNullException(nameof(formatter));
+
+            public void Emit(LogEvent logEvent)
+            {
+                if (logEvent == null)
+                    throw new ArgumentNullException(nameof(logEvent));
+
+                var testName = TestContext.CurrentContext.Test.FullName;
+
+                if (!TraceLoggers.TryGetValue(testName, out var traceLogger))
+                    throw new Exception($"Could not find trace logger for test '{testName}'");
+
+                var output = new StringWriter();
```

Comment:
Once we have the traceLogger, I wonder if it would make more sense to move the "Logging" logic into `TraceLogFileLogger.WriteLine`?

Reply:
I would prefer to keep `TraceLogFileLogger` ignorant of any other concerns besides "log this line to a file". I think the sink is a better place for the 'trim/reformat the `SourceContext` value'.